io/src/main/scala/fs2/io/file/Files.scala

```diff
@@ -0,0 +1,510 @@
+import java.nio.file.{Files => JFiles, _}
+import java.nio.file.attribute.{BasicFileAttributes, FileAttribute, PosixFilePermission}
+import java.util.stream.{Stream => JStream}
+
+import fs2.io.CollectionCompat._
+
+trait BaseFiles[F[_]] {
+
+  /**
+    * Copies a file from the source to the target path,
+    *
+    * By default, the copy fails if the target file already exists or is a symbolic link.
+    */
+  def copy(source: Path, target: Path, flags: Seq[CopyOption] = Seq.empty): F[Path]
+
+  /**
+    * Creates a new directory at the given path.
+    */
+  def createDirectory(path: Path, flags: Seq[FileAttribute[_]] = Seq.empty): F[Path]
+
+  /**
+    * Creates a new directory at the given path and creates all nonexistent parent directories beforehand.
+    */
+  def createDirectories(path: Path, flags: Seq[FileAttribute[_]] = Seq.empty): F[Path]
+
+  /**
+    * Deletes a file.
+    *
+    * If the file is a directory then the directory must be empty for this action to succeed.
+    * This action will fail if the path doesn't exist.
+    */
+  def delete(path: Path): F[Unit]
+
+  /**
+    * Like `delete`, but will not fail when the path doesn't exist.
+    */
+  def deleteIfExists(path: Path): F[Boolean]
+
+  /**
+    * Recursively delete a directory
+    */
+  def deleteDirectoryRecursively(path: Path, options: Set[FileVisitOption] = Set.empty): F[Unit]
+
+  /**
+    * Creates a stream of [[Path]]s inside a directory.
+    */
+  def directoryStream(path: Path): Stream[F, Path]
+
+  /**
+    * Creates a stream of [[Path]]s inside a directory, filtering the results by the given predicate.
+    */
+  def directoryStream(path: Path, filter: Path => Boolean): Stream[F, Path]
+
+  /**
+    * Creates a stream of [[Path]]s inside a directory which match the given glob.
+    */
+  def directoryStream(path: Path, glob: String): Stream[F, Path]
+
+  /**
+    * Checks if a file exists.
+    *
+    * Note that the result of this method is immediately outdated. If this
+    * method indicates the file exists then there is no guarantee that a
+    * subsequence access will succeed. Care should be taken when using this
+    * method in security sensitive applications.
+    */
+  def exists(path: Path, flags: Seq[LinkOption] = Seq.empty): F[Boolean]
+
+  /**
+    * Moves (or renames) a file from the source to the target path.
+    *
+    * By default, the move fails if the target file already exists or is a symbolic link.
+    */
+  def move(source: Path, target: Path, flags: Seq[CopyOption] = Seq.empty): F[Path]
+
+  /** Creates a `FileHandle` for the file at the supplied `Path`. */
+  def openPath(path: Path, flags: Seq[OpenOption]): Resource[F, FileHandle[F]]
+
+  /** Creates a `FileHandle` for the supplied `FileChannel`. */
+  def openFileChannel(channel: F[FileChannel]): Resource[F, FileHandle[F]]
+
+  /**
+    * Get file permissions as set of [[PosixFilePermission]].
+    *
+    * Note: this will only work for POSIX supporting file systems.
```

Comment:
If not posix, then does this return empty set, or fail?

Reply:
`UnsupportedOperationException`